### PR TITLE
Refactor code in src/controllers/mods.js to reduce cognitive complexity

### PR DESCRIPTION
**Pull request by mayTkhin24**
The function for listing flagged content, "modsController.flags.list", was broken down into helper functions to reduce cognitive complexity.

The helper functions are as follows:

initialize(req): Fetches necessary data like user permissions, moderated categories, and valid filter/sort options. It returns an array of these values for use in other functions.

checkAccess(isAdminOrGlobalMod, moderatedCids, req, res): Ensures the user has permission to view flags. If they don’t, it denies access; otherwise, it sets the user's moderated categories in res.locals.

parseFilters(req, filters): Extracts and validates filters from the request query. Adds filters that are present and non-empty in the query to a new filters object.

adjustFiltersForMod(filters, res): Adjusts the cid (category ID) filter to ensure the user only sees categories they moderate. If no cid is provided, it defaults to their moderated categories.

determineHasFilter(filters): Checks if any filters, other than pagination, are applied. Returns true if there are active filters.

parseSort(req, sorts): Parses the sort parameter from the request. If valid, it returns the specified sort option, or returns undefined if the sort is 'newest'.

fetchData(req, filters, sort): Fetches flags, analytics, and selected category data based on the filters and sort parameters. Uses Promise.all to handle multiple asynchronous calls.

getSelectedUsers(filters): Fetches detailed user information (username, picture) for users associated with certain filters (assignee, reporter, target user). Returns an object with user details for these filters.

### CODE REVIEW

**Review by SarimFaraz — COMMENTED** [review]
The changes are well done and implemented

**Comment by safa-amin** [comment]
the code looks good and well-structured

File: src/controllers/mods.js [hunk 1]
```diff
@@ -18,74 +18,106 @@ const helpers = require('./helpers');
...
+		selected,
+		hasFilter: hasFilter,
+		filters: filters,
+		expanded: !!(filters.assignee || filters.reporterId || filters.targetUid),
+		sort: sort || 'newest',
+		title: '[[pages:flags]]',
+		pagination: pagination.create(flagsData.page, flagsData.pageCount, req.query),
+		breadcrumbs: helpers.buildBreadcrumbs([{ text: '[[pages:flags]]' }]),
+	});
+};
```
Comment (safa-amin) [inline src/controllers/mods.js:54]: code looks well structured and easy to understand

**Review by safa-amin — COMMENTED** [review]
overall, it looks like the code efficiency has been improved. code is also neat.

**Review by mrahmed2026 — APPROVED** [review]
I approve your merge, the code seems fine to me!

**Pull request merged**